src/applications/pokemon_summary_screen/main.c

```diff
@@ -0,0 +1,2174 @@
+        *state = WaitHideBattleMoveInfo(summaryScreen);
+        break;
+    case PSS_STATE_SETUP_CONTEST_MOVE_INFO:
+        *state = WaitSetupContestMoveInfo(summaryScreen);
+        break;
+    case PSS_STATE_HIDE_CONTEST_MOVE_INFO:
+        *state = WaitHideContestMoveInfo(summaryScreen);
+        break;
+    case PSS_STATE_MOVE_SELECT:
+        *state = HandleInput_MoveSelect(summaryScreen);
+        break;
+    case PSS_STATE_MOVE_SWAP:
+        *state = HandleInput_MoveSwap(summaryScreen);
+        break;
+    case PSS_STATE_LEARN_MOVE:
+        *state = HandleInput_LearnMove(summaryScreen);
+        break;
+    case PSS_STATE_WAIT_HM_MSG_INPUT:
+        *state = WaitForHMMsgInput(summaryScreen);
+        break;
+    case PSS_STATE_SETUP_RIBBON_INFO:
+        *state = WaitSetupRibbonInfo(summaryScreen);
+        break;
+    case PSS_STATE_HIDE_RIBBON_INFO:
+        *state = WaitHideRibbonInfo(summaryScreen);
+        break;
+    case PSS_STATE_RIBBON_SELECT:
+        *state = HandleInput_RibbonSelect(summaryScreen);
+        break;
+    case PSS_STATE_SUBSCREEN_INPUT:
+        *state = HandleInput_Subscreen(summaryScreen);
+        break;
+    case PSS_STATE_SETUP_POFFIN_FEED:
+        *state = SetupPoffinFeedConditionPage(summaryScreen);
+        break;
+    case PSS_STATE_PRINT_CONTEST_STAT_MSGS:
+        *state = PrintContestStatChangeMsgs(summaryScreen);
+        break;
+    case PSS_STATE_WAIT_FINISH_POFFIN_FEED:
+        *state = WaitForPoffinFeedMsgInput(summaryScreen);
+        break;
+    case PSS_STATE_TRANSITION_OUT:
+        *state = SummaryScreenTransitionOut(summaryScreen);
+        break;
+    case PSS_STATE_WAIT_EXIT:
+        if (ScreenTransitionIsDone(summaryScreen) == TRUE) {
+            return TRUE;
+        }
+        break;
+    }
+
+    sub_0208EB14(summaryScreen);
+    PokemonSummaryScreen_DrawSheenSprites(summaryScreen);
+    sub_0208FB30(summaryScreen);
+    sub_0200C7EC(summaryScreen->gfxHandler);
+    PokemonSummaryScreen_Update3DGfx(summaryScreen);
+
+    return FALSE;
+}
+
+static int PokemonSummaryScreen_Exit(OverlayManager *ovyManager, int *state)
+{
+    PokemonSummaryScreen *summaryScreen = OverlayManager_Data(ovyManager);
+
+    SetMainCallback(NULL, NULL);
+    PokemonSummaryScreen_FreeCameraAndSpriteData(summaryScreen);
+    sub_0208EAF4(summaryScreen);
+    PokemonSummaryScreen_RemoveWindows(summaryScreen);
+    TeardownBgs(summaryScreen->bgConfig);
+    sub_0201E530();
+    VRAMTransferManager_Destroy();
+    FreeStrings(summaryScreen);
+    NARC_dtor(summaryScreen->narcPlPokeData);
+    Font_UseLazyGlyphAccess(FONT_SYSTEM);
+
+    G2_BlendNone();
+
+    OverlayManager_FreeData(ovyManager);
+    Heap_Destroy(HEAP_ID_POKEMON_SUMMARY_SCREEN);
+
+    return TRUE;
+}
+
+static void PokemonSummaryScreenVBlank(void *data)
+{
+    PokemonSummaryScreen *summaryScreen = data;
+
+    Bg_RunScheduledUpdates(summaryScreen->bgConfig);
+    sub_02008A94(summaryScreen->monSprite.spriteManager);
```

Comment:
This is probably similar to `AnimateSprites` in Emerald? Not that you need to name it at present.